core/src/main/scala/org/bitcoins/core/wallet/signer/Signer.scala

```diff
@@ -61,7 +61,28 @@ sealed abstract class P2PKSigner extends BitcoinSigner {
       hashType: HashType,
       isDummySignature: Boolean)(
       implicit ec: ExecutionContext): Future[TxSigComponent] = {
-    val spk = output.scriptPubKey
+    sign(signers,
+         output,
+         unsignedTx,
+         inputIndex,
+         hashType,
+         isDummySignature,
+         p2wshOverride = None)
+  }
+
+  def sign(
+      signers: Seq[Sign],
+      output: TransactionOutput,
+      unsignedTx: Transaction,
+      inputIndex: UInt32,
+      hashType: HashType,
+      isDummySignature: Boolean,
+      p2wshOverride: Option[(TxSigComponent, ScriptPubKey)])(
```

Comment:
Hoping to find a nice general type for whatever this will be in the future. `timeLockOverride` will have just a `ScriptPubKey` while `OP_IF` will require an override with an SPK as well, actually now that I am writing this down I can think of the perfect type, an ADT :) will go do this now

Reply:
Done :)